java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/model/BulkExportQuery.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.model;
+
+import java.util.Objects;
+
+/**
+ * A request for a bulk export on a given table.
+ */
+public class BulkExportQuery {
+    private final String exportId;
+    private final String tableId;
+    private final String tableName;
+
+    private BulkExportQuery(Builder builder) {
+        tableId = requireNonNull(builder.tableId, builder, "tableId field must be provided");
+        tableName = requireNonNull(builder.tableName, builder, "tableName field must be provided");
```

Comment:
I think the only reason to have both the table ID and name would be to allow a client to send the table name without knowing the ID, or to send the table ID if it is known.

I think we need to allow just one of these fields to be set at a time. If both are set, we could either just take the table ID, or throw an exception. This is enforcing that both must be set, which I think should never actually happen.

Reply:
It now throes an exception if both are missing or provided.